src/test/java/guitests/PanelRenameTest.java

```diff
@@ -0,0 +1,120 @@
+package guitests;
+
+import javafx.scene.input.KeyCode;
+
+import org.junit.Test;
+import org.loadui.testfx.utils.FXTestUtils;
+
+import prefs.Preferences;
+import ui.UI;
+import ui.components.PanelNameTextField;
+import util.PlatformEx;
+import util.events.ShowRenamePanelEvent;
+
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+public class PanelRenameTest extends UITest {
+
+    public static final int EVENT_DELAY = 500;
+
+    @Override
+    public void launchApp() {
+        FXTestUtils.launchApp(TestUI.class, "--testconfig=true", "--bypasslogin=true");
+    }
+
+    @Test
+    public void panelRenameTest() {
+        
+        // Test for saving panel information with corner cases
+        
+        // maximize
+        press(KeyCode.CONTROL).press(KeyCode.X).release(KeyCode.X).release(KeyCode.CONTROL);
+        sleep(EVENT_DELAY);
+
+        // Testing rename cancel
+        PlatformEx.runAndWait(() -> UI.events.triggerEvent(new ShowRenamePanelEvent(0)));
+        push(KeyCode.ESCAPE);
+        sleep(EVENT_DELAY);
+        
+        press(KeyCode.CONTROL).press(KeyCode.P).release(KeyCode.P).release(KeyCode.CONTROL);
+        
+        // Testing panel names between whitespaces
+        PlatformEx.runAndWait(() -> UI.events.triggerEvent(new ShowRenamePanelEvent(1)));
+        type("   Renamed panel  ");
+        push(KeyCode.ENTER);
+        sleep(EVENT_DELAY);
+        
+        press(KeyCode.CONTROL).press(KeyCode.P).release(KeyCode.P).release(KeyCode.CONTROL);
+        
+        // Testing empty panel name
+        PlatformEx.runAndWait(() -> UI.events.triggerEvent(new ShowRenamePanelEvent(2)));
+        push(KeyCode.BACK_SPACE);
+        push(KeyCode.ENTER);
+        sleep(EVENT_DELAY);
+        
+        press(KeyCode.CONTROL).press(KeyCode.P).release(KeyCode.P).release(KeyCode.CONTROL);
+
+        // Testing typing excessive panel name
+        PlatformEx.runAndWait(() -> UI.events.triggerEvent(new ShowRenamePanelEvent(3)));
+        sleep(EVENT_DELAY);
+        type("1234567890123456789012345678901234567890");
```

Comment:
How long is this and why was that value chosen? Is it a boundary value? If not, it should be. If the max length is 20, you should test 19,20,21. It may be useful to define a utility method that can give you a random string of a specified length. e.g. `StringHelper.getRandomString(PANEL_NAME_MAX_LENGTH+1)`